pkg/sloop/store/typed/eventcounttablegen.go

```diff
@@ -270,3 +271,86 @@ func (t *ResourceEventCountsTable) getLastMatchingKeyInPartition(txn badgerwrap.
 	}
 	return false, &EventCountKey{}, nil
 }
+
+//todo: add unit tests
+func (t *ResourceEventCountsTable) RangeReadPerPartition(txn badgerwrap.Txn, keyPrefix *EventCountKey,
+	valPredicateFn func(*ResourceEventCounts) bool, startTime time.Time, endTime time.Time) (map[EventCountKey]*ResourceEventCounts, RangeReadStats, error) {
+	resources := map[EventCountKey]*ResourceEventCounts{}
+
+	stats := RangeReadStats{}
+	before := time.Now()
+
+	partitionList, err := t.GetPartitionsFromTimeRange(txn, startTime, endTime)
+	stats.PartitionCount = len(partitionList)
+	if err != nil {
+		return resources, stats, errors.Wrapf(err, "failed to get partitions from table:%v, from startTime:%v, to endTime:%v", t.tableName, startTime, endTime)
+	}
+
+	tablePrefix := "/" + t.tableName + "/"
+	for _, currentPartition := range partitionList {
+		var seekStr string
+
+		// when keyPrefix does not have partition info(which implies it may not contain kind,namespace,and etc), we seek from /tableName/currentPartition/
+		if keyPrefix.PartitionId == "" {
```

Comment:
Can we instead check if keyPrefix is a nil ptr?

Reply:
sure, we can do that.